index.js

```diff
@@ -199,18 +211,27 @@ P.prototype._loadQuery = async function loadQuery(bbox, q) {
   self._recalc()
 }
 
-P.prototype._recalc = function() {
+P.prototype._recalc = function(fromScheduled) {
   var self = this
+  if (!fromScheduled && !self._recalcTimer) {
+    self._debug('something is calling _recalc() directly and nothing is scheduled')
+  } else if (!fromScheduled && self._recalcTimer) {
+    self._debug('something is calling _recalc() directly while already scheduled')
```

Comment:
One simple optimization is just to add a `return` statement here since `_recalc()` is about to be called anyway.